### PR TITLE
Issue/navigate safely

### LINKED ISSUE #2400: Order refund navigation crash
[This Sentry crash](https://sentry.io/organizations/a8c/issues/1129511887/events/latest/?project=1459556&query=is%3Aunresolved) was incorrectly attached to #1719. That issue was a navigation crash in order fulfillment, but this one is in order refund.

```
IllegalArgumentException
navigation destination com.woocommerce.android:id/
action_issueRefundFragment_to_refundItemsPickerDialog is unknown to this NavController
```

I was able to reproduce the crash by rapidly tapping twice here:

Screenshot_1588963074

### PR DESCRIPTION
**Pull request by nbradbury**
Fixes #2400, #2074, #2400, and #1719 - we have a ton of Sentry reports of crashes caused by navigating to the same destination twice after users rapidly click the same view.

We've tried various ways to fix these, but the crashes are still happening. I've opened this PR which adds a `NavController` extension that enables safely navigating and avoiding these crashes.

One way to initially test this is to run `develop` then rapidly double-click here in order refund:

Screenshot_1588963074

That will crash the app. Now run this branch and see the crash is gone.

@anitaa1990, @AmandaRiu, @0nko - I've asked all three of you to give this a spin since it changes navigation throughout the app. It would be good to get this merged soon since it touches so many files.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/NavController.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * Prevents crashes caused by rapidly double-clicking views which navigate to the same
+ * destination twice
+ */
+fun NavController.navigateSafely(directions: NavDirections) {
+    currentDestination?.getAction(directions.actionId)?.let { navigate(directions) }
+}
+
+fun NavController.navigateSafely(@IdRes resId: Int) {
+    if (currentDestination?.id == resId == false) {
```
Comment (AmandaRiu) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/NavController.kt:16]: np: could be simplified with:

```
if (currentDestination?.id != resId) {
```

**Comment by AmandaRiu** [comment]
@nbradbury this one should probably have release notes added as well.

**Comment by nbradbury** [comment]
@AmandaRiu Ready for another round!

**Comment by AmandaRiu** [comment]
Since this was already approved by @anitaa1990 , @0nko and myself, I'm going to go ahead and merge it 👍

**Comment by AmandaRiu** [comment]
Note: Another crash has been logged in Sentry for the 4.3-rc-1 build.

**Pull request merged**